execution.bs

```diff
@@ -3173,67 +3196,91 @@ Insert this section as a new subclause between Header `<stop_token>` synopsis <b
 
-    2. If `t.stop_possible()` evaluates to `false` then, if `u`, references the same logical shared stop state, `u.stop_possible()` shall also subsequently evaluate to `false` and `u.stop_requested()` shall also subsequently evaluate to `false`.
+    2. If `t.stop_requested()` evaluates to `true` then, if `u` references the same logical shared stop state, `u.stop_requested()` shall also subsequently evaluate to `true` and `u.stop_possible()` shall also subsequently evaluate to `true`.
 
-    3. If `t.stop_requested()` evaluates to `true` then, if `u`, references the same logical shared stop state, `u.stop_requested()` shall also subsequently evaluate to `true` and `u.stop_possible()` shall also subsequently evaluate to `true`.
+3. Let `t` and `u` be distinct, valid objects of type `T` and let `init` be an
+    object of type `Initializer`. Then for some type `CB`, the type `T` models
+    `stoppable_token_for<CB, Initializer>` only if:
 
-    4. Given a callback-type, CB, and a callback-initializer argument, `init`, of type `Initializer` then constructing an instance, `cb`, of type `T::callback_type<CB>`, passing `t` as the first argument and `init` as the second argument to the constructor, shall,
-        if `t.stop_possible()` is `true`, construct an instance, `callback`, of type `CB`, direct-initialized with `init`, and register callback with `t`’s shared stop state such that callback will be invoked with an empty argument list if a stop request is made on
-        the shared stop state.
+    1. Constructing an instance, `cb`, of type `T::callback_type<CB>`, passing
+        `t` as the first argument and `init` as the second argument to the
+        constructor, shall, if `t.stop_possible()` is `true`, construct an
+        instance, `callback`, of type `CB`, direct-initialized with `init`,
+        and register callback with `t`'s shared stop state such that `callback`
+        will be invoked with an empty argument list if a stop request is made
+        on the shared stop state.
 
-        1. If `t.stop_requested()` is `true` at the time callback is registered then callback may be invoked immediately inline inside the call to `cb`’s constructor.
+        1. If `t.stop_requested()` is `true` at the time `callback` is
+            registered then `callback` may be invoked immediately inline inside
+            the call to `cb`'s constructor.
 
-        2. If callback is invoked then, if `u` references the same shared stop state as `t`, an evaluation of `u.stop_requested()` will be `true` if the beginning of the invocation of callback strongly-happens-before the evaluation of `u.stop_requested()`.
+        2. If `callback` is invoked then, if `u` references the same shared stop
+            state as `t`, an evaluation of `u.stop_requested()` will be `true`
+            if the beginning of the invocation of `callback`
+            strongly-happens-before the evaluation of `u.stop_requested()`.
```

Comment:
I wonder if this could be relaxed to `happens-before` here rather than `strongly-happens-before`.

Might be worth a check-in with SG1 people.

Reply:
I quake at the thought of bringing _any_ questions to SG1. :grimacing:

Reply:
I asked on the sg1 reflector. I interpret Hans' answer to mean that "strongly happens before" is correct here.